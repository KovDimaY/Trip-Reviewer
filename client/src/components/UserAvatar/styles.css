.fit-parent {
    width: 100%;
    height: 100%;
}
.user-avatar-container .user-avatar-spinner {
    line-height: 100px;
    margin-top: 30px;
    text-align: center;
    vertical-align: middle;
}
.user-avatar-container .avatar-picture img {
    max-width: 200px;
    max-height: 200px;
    box-shadow: 1px 3px 15px black;
    border: 1px solid;
    object-fit: cover;
}