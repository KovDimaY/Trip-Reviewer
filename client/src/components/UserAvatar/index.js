import React, { PureComponent } from 'react';
import Loader from '../Loaders/PulseLoader';

import { firebase } from '../../firebase';
import './styles.css';

class UserAvatar extends PureComponent {
    state = {
      src: '',
      isLoading: true,
    };

    componentWillMount() {
      this.updateImage(this.props);
    }

    componentWillReceiveProps(nextProps) {
      this.updateImage(nextProps);
    }

    updateImage(props) {
      const { filename } = props;

      if (filename) {
        firebase.storage().ref('avatars')
          .child(filename).getDownloadURL()
          .then(this.handleGetUrlSuccess);
      } else {
        this.setState({ src: '/images/avatar.png', isLoading: false });
      }
    }

    handleGetUrlSuccess = (url) => {
      this.setState({ src: url, isLoading: false });
    }

    renderWithLoading() {
      const { isLoading, src } = this.state;

      if (isLoading) {
        return <Loader className="user-avatar-spinner" color="grey" size="16px" margin="4px" />;
      }
      return (
        <div className="avatar-picture">
          <img alt="avatar" src={src} />
        </div>
      );
    }

    render() {
      return (
        <div className="user-avatar-container">
          { this.renderWithLoading() }
        </div>
      );
    }
}

export default UserAvatar;
