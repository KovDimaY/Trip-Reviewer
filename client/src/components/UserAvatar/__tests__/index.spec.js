import React from 'react';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';
 
import UserAvatar from './../../UserAvatar';

jest.mock('./../../Loaders/PulseLoader', () => ('Loader'));
jest.mock('./../../../firebase', () => ({
    firebase: {
        storage: jest.fn(() => ({
            ref: jest.fn(() => ({
                child: jest.fn(() => ({
                    getDownloadURL: jest.fn(() => ({
                        then: jest.fn()
                    }))
                }))
            }))
        }))
    }
}));


const mockComponent = (props) => {
    return (
        <UserAvatar {...props} />
    );
};
 
describe('<UserAvatar />', () => {
    it('should render component with no filename', () => {
        const props = {
            filename: ''
        };
        const tree = create(mockComponent(props)).toJSON();
        
        expect(tree).toMatchSnapshot();
    });

    it('should render component with filename', () => {
        const props = {
            filename: 'filename'
        };
        const tree = create(mockComponent(props)).toJSON();
        
        expect(tree).toMatchSnapshot();
    });

    it('should change state when handleGetUrlSuccess is called', () => {
        const instance = shallow(mockComponent({})).instance();
    
        instance.handleGetUrlSuccess('url');
    
        expect(instance.state.src).toEqual('url');
        expect(instance.state.isLoading).toEqual(false);
    });

    it('should call updateImage when componentWillReceiveProps is called', () => {
        const updateImage = jest.fn();
        const instance = shallow(mockComponent({})).instance();
        
        instance.updateImage = updateImage;
        instance.componentWillReceiveProps('test');
    
        expect(updateImage).toHaveBeenCalledWith('test');
    });
});