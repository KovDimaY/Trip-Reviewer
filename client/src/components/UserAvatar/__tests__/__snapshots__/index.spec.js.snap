// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserAvatar /> should render component with filename 1`] = `
<div
  className="user-avatar-container"
>
  <Loader
    className="user-avatar-spinner"
    color="grey"
    margin="4px"
    size="16px"
  />
</div>
`;

exports[`<UserAvatar /> should render component with no filename 1`] = `
<div
  className="user-avatar-container"
>
  <div
    className="avatar-picture"
  >
    <img
      alt="avatar"
      src="/images/avatar.png"
    />
  </div>
</div>
`;
